Buildings/Fluid/Storage/Plant/TankBranch.mo

```diff
@@ -0,0 +1,316 @@
+          -60},{-100,-60}}, color={0,127,255}));
+  connect(port_2, port_2)
+    annotation (Line(points={{100,-60},{100,-60}}, color={0,127,255}));
+  connect(sen_m_flow.port_a, valCha.port_a) annotation (Line(points={{-40,30},{-60,
+          30},{-60,0},{30,0},{30,-80},{20,-80}}, color={0,127,255}));
+  connect(pasValDis.port_b, sen_m_flow.port_a) annotation (Line(points={{20,-50},
+          {30,-50},{30,0},{-60,0},{-60,30},{-40,30}}, color={0,127,255}));
+  connect(valDis.port_b, sen_m_flow.port_a) annotation (Line(points={{20,-32},{30,
+          -32},{30,0},{-60,0},{-60,30},{-40,30}}, color={0,127,255}));
+  connect(sen_m_flow.port_a, port_3) annotation (Line(points={{-40,30},{-60,30},
+          {-60,100}},          color={0,127,255}));
+  connect(preDroTan.port_b, port_4) annotation (Line(points={{60,30},{70,30},{70,
+          86},{60,86},{60,100}},
+                         color={0,127,255}));
+  connect(valCha.y_actual, yValCha_actual)
+    annotation (Line(points={{5,-73},{-20,-73},{-20,-110}}, color={0,0,127}));
+  connect(valDis.y_actual, yValDis_actual)
+    annotation (Line(points={{15,-25},{40,-25},{40,-110}}, color={0,0,127}));
+  connect(valCha.y, yValCha)
+    annotation (Line(points={{10,-68},{-40,-68},{-40,-110}}, color={0,0,127}));
+  connect(yValDis, valDis.y)
+    annotation (Line(points={{60,-110},{60,-20},{10,-20}}, color={0,0,127}));
+  connect(pum.y, yPum) annotation (Line(points={{-70,-28},{-70,-20},{-110,-20}},
+        color={0,0,127}));
+  annotation (Icon(coordinateSystem(preserveAspectRatio=false, extent={{-100,-100},
+            {100,100}}),       graphics={Line(
+          points={{-30,-92},{30,-92}},
+          color={28,108,200},
+          pattern=LinePattern.Dash,
+          visible=allowRemoteCharging), Polygon(
+          points={{-30,-92},{-10,-86},{-10,-98},{-30,-92}},
+          fillColor={28,108,200},
+          fillPattern=FillPattern.Solid,
+          pattern=LinePattern.None,
+          visible=allowRemoteCharging),
+        Line(points={{-30,-72},{30,-72}},color={28,108,200}),
+        Polygon(
+          points={{30,-72},{10,-66},{10,-78},{30,-72}},
+          fillColor={28,108,200},
+          fillPattern=FillPattern.Solid,
+          pattern=LinePattern.None),
+        Line(points={{-100,-60},{-40,-60},{-40,-30},{0,-30},{0,30},{40,30},{40,-60},
+              {100,-60}}, color={0,0,0}),
+        Rectangle(
+          extent={{-20,40},{20,-40}},
+          lineColor={0,0,0},
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid),
+        Line(points={{-40,100},{-40,-30}}, color={0,0,0}),
+        Line(points={{40,100},{40,30}}, color={0,0,0}),
+        Text(
+          extent={{-62,-124},{62,-100}},
+          textColor={0,0,127},
+          textString="%name")}),                                 Diagram(
+        coordinateSystem(preserveAspectRatio=false, extent={{-100,-100},{100,100}})),
+    Documentation(info="<html>
```

Comment:
Update the documentation to only reflect what this model contains. Example:

This model represents a thermal energy storage system composed by a stratified water tank connected to a pump and valve group that allows discharging and local, remote or local and remote charging depending on the value of `chaTyp`.